3scale/porta
# THREESCALE-765: Brute force protection for the admin login screen

**What this PR does / why we need it**:

Add a Recaptcha to the admin portal login screen.

This PR includes a new screen under **Account Settings -> Integrate -> Bot Protection** to enable/disable the Recaptcha.

**Which issue(s) this PR fixes** 

https://issues.redhat.com/browse/THREESCALE-765

**Verification steps** 

1. Go to **Account Settings -> Integrate -> Bot Protection**
2. Select **reCAPTCHA**
3. Go to the provider login screen
4. You should see the text
```
This site is protected by reCAPTCHA and the Google Privacy Policy and Terms of Service apply.
```
5. The login screen should work as usual: return an error when the credentials are invalid and log in when the credentials are valid
6. Set `RECAPTCHA_MIN_BOT_SCORE` to `0.99` and restart the server
7. You should always be detected as a bot, no matter whether the credentials are valid or not.

**Special notes for your reviewer**:

Some references I took to write the React component:

- https://stackoverflow.com/questions/34424845/adding-script-tag-to-react-jsx
- https://w11i.me/recaptcha-v3-react
- https://stackoverflow.com/questions/53832882/react-and-recaptcha-v3
- https://developers.google.com/recaptcha/docs/v3#programmatically_invoke_the_challenge
- https://developers.google.com/recaptcha/docs/faq#id-like-to-hide-the-recaptcha-badge.-what-is-allowed

Please, @josemigallas @lvillen review the UI part, it's probably not aligned with design guidelines or not well integrated with the rest of the UI code base.
